drivers/timer/arm_arch_timer.c

```diff
@@ -98,6 +98,22 @@ void sys_clock_set_timeout(int32_t ticks, bool idle)
 
 	k_spinlock_key_t key = k_spin_lock(&lock);
 	uint64_t curr_cycle = arm_arch_timer_count();
+
+	/*
+	 * Continuously reading the cntvct_el0 within a short time interval may
+	 * get the same value so that it may cause a side effect when setting
+	 * the next timeout interrupt especially for the board with a
+	 * low-frequency clock.
+	 * For example, On FVP platform, When a timer compare interrupt occurs,
+	 * we got 530000 from cntvct_el0 in the arm_arch_timer_compare_isr
+	 * function and we may also get 530000 from cntvct_el0 here. This will
+	 * cause the req_cycle to round up to the wrong value and cause the
+	 * timer compare interrupt to occur earlier than expected.
+	 */
+	if (curr_cycle == last_cycle) {
+		curr_cycle += 1;
+	}
```

Comment:
From https://github.com/zephyrproject-rtos/zephyr/issues/46366:

> p.s. This also fails on `fvp_baser_aemv8r`; interestingly, it does not fail on `fvp_baser_aemv8r_aarch32`.

So how can we explain this working properly on `fvp_baser_aemv8r_aarch32`?

Reply:
Let me confirm it.